feat: add `stats/incr/nanpcorr`

Girish-Garg opened the pull request:
Resolves #5616.

## Description

> What is the purpose of this pull request?

This pull request:

-   add incremental Pearson correlation coefficient accumulator that ignores NaN values by making a wrapper over pre-existing incrpcorr function
- Here are all the checks:

test:
![image](https://github.com/user-attachments/assets/8768eba2-057b-4489-8b11-326096c81b04)

example:
![image](https://github.com/user-attachments/assets/c337fcde-2843-488d-b046-c8a7087ae316)

benchmark:
![image](https://github.com/user-attachments/assets/24e762b4-5a8a-40eb-9a41-eb97af0d1572)

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5616  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Girish-Garg committed:
feat: add incremental Pearson correlation coefficient accumulator that ignores NaN values

Girish-Garg commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Girish-Garg committed:
Merge branch 'stdlib-js:develop' into feature/nanpcorr

Girish-Garg committed:
chore: fixed linting and clean-up

Girish-Garg committed:
chore: changed example file

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanpcorr/lib/index.js:
```diff
@@ -0,0 +1,57 @@
+* @module @stdlib/stats/incr/nanpcorr
+*
+* @example
+* var incrpcorr = require( '@stdlib/stats/incr/nanpcorr' );
```
The variable name in the example should match the function being exported. Should be `incrnanpcorr` instead of `incrpcorr`.

```suggestion
* var incrnanpcorr = require( '@stdlib/stats/incr/nanpcorr' );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanpcorr/lib/main.js:
```diff
@@ -0,0 +1,205 @@
+* // returns ~-1.0
+*
+* @example
+* var accumulator = incrpcorr( 2.0, -3.0 );
```
The second `@example` references the wrong function name - should be `incrnanpcorr` rather than `incrpcorr`.

```suggestion
* var accumulator = incrnanpcorr( 2.0, -3.0 );
```

Planeshifter replied:
Wrong function name. Should be `incrnanpcorr`.

```suggestion
* var accumulator = incrnanpcorr( 2.0, -3.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanpcorr/lib/main.js:
```diff
@@ -0,0 +1,205 @@
+	* @private
+	* @param {number} [x] - new value
+	* @param {number} [y] - new value
+	* @returns {(number|null)} sample absolute correlation coefficient or null
```
Small typo here - the word "absolute" shouldn't be in this description. This function returns the Pearson correlation coefficient (which can be positive or negative), not the absolute correlation.

```suggestion
	* @returns {(number|null)} sample correlation coefficient or null
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanpcorr/docs/types/index.d.ts:
```diff
@@ -0,0 +1,69 @@
+* @returns accumulator function
+*
+* @example
+* var accumulator = incrpcorr( 2.0, -3.0 );
```
Should be `incrnanpcorr` instead of `incrpcorr`.

```suggestion
* var accumulator = incrnanpcorr( 2.0, -3.0 );
```

Planeshifter replied:
Wrong function name. Should be `incrnanpcorr`. Same on line 50.

```suggestion
* var accumulator = incrnanpcorr( 2.0, -3.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanpcorr/docs/types/index.d.ts:
```diff
@@ -0,0 +1,69 @@
+* @returns accumulator function
+*
+* @example
+* var accumulator = incrpcorr();
```
Should be `incrnanpcorr` instead of `incrpcorr`.

```suggestion
* var accumulator = incrnanpcorr();
```

Planeshifter reviewed (changes requested):
Left some initial comments. Thanks!

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanpcorr/package.json:
```diff
@@ -0,0 +1,71 @@
+{
+  "name": "@stdlib/stats/incr/nanpcorr",
+  "version": "0.0.0",
+  "description": "Compute a sample Pearson product-moment correlation coefficient, ignoring `NaN` values.",
```
Don't wrap `NaN` in backticks in `package.json` descriptions.

```suggestion
  "description": "Compute a sample Pearson product-moment correlation coefficient, ignoring NaN values.",
```

Planeshifter committed:
chore: address review feedback and update copyright years

- fix `incrnanpcorr` function name in examples
- remove stray "absolute" in accumulator description
- use `format` for benchmark name string concatenation

Planeshifter commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/nanpcorr

Planeshifter committed:
chore: address review feedback

- remove unused `isnan` import from README example
- clear hand-populated See Also/related-links sections for auto-population
